File: 프로그래머스/lv1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int fct(vector<int> &array, vector<int> commands) {
    vector<int> sub(array.begin() + commands[0] - 1, array.begin() + commands[1]);
    sort(sub.begin(), sub.end());
    return sub[commands[2] - 1];
}

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (int idx = 0; idx < commands.size(); idx++)
        answer.push_back(fct(array, commands[idx]));
    return answer;
}

File: 프로그래머스/Lv.1/42748. K번째수/K번째수.java
import java.util.Arrays;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int len = commands.length;
        int[] answer = new int[len];
        for (int i = 0; i < len; i++) {
            int sublen = commands[i][1] - commands[i][0] + 1;
            int[] tmp = new int[sublen];
            System.arraycopy(array, commands[i][0] - 1, tmp, 0, sublen);
            Arrays.sort(tmp);
            answer[i] = tmp[commands[i][2] - 1];      
        }
        return answer;
    }
}